cvat/apps/events/handlers.py

```diff
@@ -412,16 +413,50 @@ def handle_delete(scope, instance, store_in_deletion_cache=False, **kwargs):
     )
 
 
-def handle_annotations_change(instance, annotations, action, **kwargs):
+def handle_annotations_change(instance: Job, annotations, action, **kwargs):
     def filter_data(data):
-        filtered_data = {
+        return {
             "id": data["id"],
         }
 
-        return filtered_data
+    in_mem_counter = TracksCounter()
+    in_mem_counter.load_tracks_from_job_collection(
+        instance.id, instance.segment.stop_frame, {"tracks": annotations.get("tracks", [])}
+    )
+
+    ib_db_counter = TracksCounter()
+    if action == "update" and annotations.get("tracks", []):
+        ib_db_counter.load_tracks_from_db(
+            parent_labeledtrack_qs_filter=lambda x: x.filter(
+                pk__in=(track["id"] for track in annotations["tracks"])
+            ),
+            child_labeledtrack_qs_filter=lambda x: x.filter(
+                parent_id__in=(track["id"] for track in annotations["tracks"])
```

Comment:
Typically such queries with long WHERE id in ... lists work quite slow. Consider changing the logic to filter fetched annotations after, in the python code.

Reply:
I do not think there will be many values `in__(ids)` in practice. Usually users "updates" just several tracks.
I do not think it would be efficient on each save retrieve all tracks from the database.